sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/jsonExpressions.scala

```diff
@@ -149,7 +149,7 @@ case class GetJsonObject(json: Expression, path: Expression)
 
     if (parsed.isDefined) {
       try {
-        Utils.tryWithResource(jsonFactory.createParser(jsonStr.getBytes)) { parser =>
+        Utils.tryWithResource(jsonFactory.createParser(jsonStr.toString)) { parser =>
```

Comment:
I think we should catch the exception below rather than introducing unnecessary string instance apparently. (Sorry, I got confused it was `from_json`/`to_json`).